### PR TITLE
vimplugins: add localvimrc, golden-ratio, and vim-rails

### PR DESCRIPTION
**Pull request by pseudomuto**
###### Motivation for this change

Update vimPlugins to include the following packages:

* https://github.com/embear/vim-localvimrc
* https://github.com/roman/golden-ratio
* https://github.com/tpope/vim-rails

###### Things done

Added packages with the following command:

```
GITHUB_API_TOKEN=<REDACTED> ./update.py \
  --add "embear/vim-localvimrc" \
  --add "roman/golden-ratio" \
  --add "tpope/vim-rails"
```

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by pseudomuto** [comment]
Rebased to handle conflicts

**Comment by pseudomuto** [comment]
@jonringer I've rebased this now that #93686 has been merged and I could run `./update.py`.

File: pkgs/misc/vim-plugins/vim-plugin-names [hunk 1]
```diff
@@ -96,6 +96,7 @@ elixir-editors/vim-elixir
 elmcast/elm-vim
 elzr/vim-json
 embear/vim-localvimrc
+embear/vim-localvimrc
```
Comment (jonringer) [inline pkgs/misc/vim-plugins/vim-plugin-names:99]: this file was already added, please drop commit
`git rebase -i` makes this easy

**Comment by pseudomuto** [comment]
Removed localvimrc plugin from this PR since it's been added already

**Comment by jonringer** [comment]
@GrahamcOfBorg eval

**Review by jonringer — APPROVED** [review]
not sure why ofborg is failing to eval
LGTM

```
https://github.com/NixOS/nixpkgs/pull/93251
37 packages built:
vimPlugins.Tagbar vimPlugins.airline vimPlugins.ale vimPlugins.calendar vimPlugins.coc-fzf vimPlugins.coc-spell-checker vimPlugins.completion-nvim vimPlugins.conjure vimPlugins.cpsm vimPlugins.defx-icons vimPlugins.denite vimPlugins.fugitive vimPlugins.git-messenger-vim vimPlugins.golden-ratio vimPlugins.julia-vim vimPlugins.lexima-vim vimPlugins.lightline-bufferline vimPlugins.ncm2-neosnippet vimPlugins.neoformat vimPlugins.neosnippet vimPlugins.neuron-vim vimPlugins.nvim-treesitter vimPlugins.quick-scope vimPlugins.table-mode vimPlugins.vim-bufkill vimPlugins.vim-eunuch vimPlugins.vim-floaterm vimPlugins.vim-plug vimPlugins.vim-ps1 vimPlugins.vim-rails vimPlugins.vim-ruby vimPlugins.vim-snippets vimPlugins.vim-test vimPlugins.vim-visual-multi vimPlugins.vim-vsnip vimPlugins.vimtex vimPlugins.vista-vim
```

**Pull request merged**